### ISSUE TITLE
Nodes panicked at 'supplied instant is later than self'

### ISSUE THREAD

**Issue by trietnguyen267** [cennznet/cennznet#181]
Running nodes sometime got panicked across regions which seems to be related to timestamp

```
│ az-sg-cennznet-v-b 2020-04-01 19:56:36 :zzz:  Idle (14 peers), best: #11290 (0x7988…7a57), finalized #11284 (0xe62b…fee9), ⬇ 5.0kiB/s ⬆ 29.9kiB/s                                                                                                                             │
│ az-sg-cennznet-v-b 2020-04-01 19:56:39 :raised_hands:  Starting consensus session on top of parent 0x7988ae4066d456789173b818eddefdffa38f8b5b980ead998c18047db9557a57                                                                                                                  │
│ az-sg-cennznet-v-b ====================                                                                                                                                                                                                                                   │
│ az-sg-cennznet-v-b Version: 1.0.0-c17def9-x86_64-linux-gnu                                                                                                                                                                                                                │
│ az-sg-cennznet-v-b    0: sp_panic_handler::set::{{closure}}                                                                                                                                                                                                               │
│ az-sg-cennznet-v-b    1: std::panicking::rust_panic_with_hook                                                                                                                                                                                                             │
│ az-sg-cennznet-v-b              at src/libstd/panicking.rs:476                                                                                                                                                                                                            │
│ az-sg-cennznet-v-b    2: rust_begin_unwind                                                                                                                                                                                                                                │
│ az-sg-cennznet-v-b              at src/libstd/panicking.rs:380                                                                                                                                                                                                            │
│ az-sg-cennznet-v-b    3: core::panicking::panic_fmt                                                                                                                                                                                                                       │
│ az-sg-cennznet-v-b              at src/libcore/panicking.rs:85                                                                                                                                                                                                            │
│ az-sg-cennznet-v-b    4: core::option::expect_failed                                                                                                                                                                                                                      │
│ az-sg-cennznet-v-b              at src/libcore/option.rs:1191                                                                                                                                                                                                             │
│ az-sg-cennznet-v-b    5: core::option::Option<T>::expect                                                                                                                                                                                                                  │
│ az-sg-cennznet-v-b              at rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/option.rs:347                                                                                                                                                               │
│ az-sg-cennznet-v-b       std::time::Instant::duration_since                                                                                                                                                                                                               │
│ az-sg-cennznet-v-b              at src/libstd/time.rs:263                                                                                                                                                                                                                 │
│ az-sg-cennznet-v-b       <std::time::Instant as core::ops::arith::Sub>::sub                                                                                                                                                                                               │
│ az-sg-cennznet-v-b              at src/libstd/time.rs:390                                                                                                                                                                                                                 │
│ az-sg-cennznet-v-b    6: sc_basic_authorship::basic_authorship::ProposerInner<B,Block,C,A>::propose_with                                                                                                                                                                  │
│ az-sg-cennznet-v-b    7: core::ops::function::FnOnce::call_once{{vtable.shim}}                                                                                                                                                                                            │
│ az-sg-cennznet-v-b    8: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once                                                                                                                                                                              │
│ az-sg-cennznet-v-b    9: __rust_maybe_catch_panic                                                                                                                                                                                                                         │
│ az-sg-cennznet-v-b              at src/libpanic_unwind/lib.rs:86                                                                                                                                                                                                          │
│ az-sg-cennznet-v-b   10: std::sys_common::backtrace::__rust_begin_short_backtrace                                                                                                                                                                                         │
│ az-sg-cennznet-v-b   11: __rust_maybe_catch_panic                                                                                                                                                                                                                         │
│ az-sg-cennznet-v-b              at src/libpanic_unwind/lib.rs:86                                                                                                                                                                                                          │
│ az-sg-cennznet-v-b   12: core::ops::function::FnOnce::call_once{{vtable.shim}}                                                                                                                                                                                            │
│ az-sg-cennznet-v-b   13: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once                                                                                                                                                                              │
│ az-sg-cennznet-v-b              at rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/liballoc/boxed.rs:1015                                                                                                                                                              │
│ az-sg-cennznet-v-b   14: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once                                                                                                                                                                              │
│ az-sg-cennznet-v-b              at rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/liballoc/boxed.rs:1015                                                                                                                                                              │
│ az-sg-cennznet-v-b       std::sys_common::thread::start_thread                                                                                                                                                                                                            │
│ az-sg-cennznet-v-b              at src/libstd/sys_common/thread.rs:13                                                                                                                                                                                                     │
│ az-sg-cennznet-v-b       std::sys::unix::thread::Thread::new::thread_start                                                                                                                                                                                                │
│ az-sg-cennznet-v-b              at src/libstd/sys/unix/thread.rs:80                                                                                                                                                                                                       │
│ az-sg-cennznet-v-b   15: start_thread                                                                                                                                                                                                                                     │
│ az-sg-cennznet-v-b   16: __clone                                                                                                                                                                                                                                          │
│ az-sg-cennznet-v-b Thread 'tokio-blocking-driver' panicked at 'supplied instant is later than self', src/libstd/time.rs:263                                                                                                                                               │
│ az-sg-cennznet-v-b This is a bug. Please report it at:                                                                                                                                                                                                                    │
│ az-sg-cennznet-v-b     https://github.com/cennznet/cennznet/issues/new                                                                                                                                                                                                    │
│ <STREAM> closed 

```

The issue replicated itself quite frequently on same/different node. After restarted all of them can be synced and run normally.

**Comment by jordy25519** [comment]
Upstream fix: https://github.com/paritytech/substrate/pull/5493

**Comment by trietnguyen267** [comment]
Ref to original issue on substrate
https://github.com/paritytech/substrate/issues/4949

**Comment by jordy25519** [comment]
closed via #233
